### PR TITLE
Add auto-resize functionality

### PR DESCRIPTION
**Pull request by kyl191**
Images now get resized when uploaded to a bucket!

~600ms for the actual resizing with a 768MB container, ~2 sec for the complete activation, download/resize/upload/delete cycle.

Might have to look at the cross posting function, add a bit of delay to make sure the resized file will be available.

### CODE REVIEW

File: ResizeImage.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        bucket = record['s3']['bucket']['name']
+        # S3 keys are urlencoded in event notifications
+        # https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
+        key = unquote_plus(record['s3']['object']['key'])
+        size = record['s3']['object']['size']
+        log.debug("Got new file: %s:%s, size %d", bucket, key, size)
+        original = get_file(bucket, key)
+        resized = resize_image(original)
+        put_file(resized,
+                 os.environ.get('DEST_BUCKET', 'calligre-images'),
```
Comment (TheKevJames) [inline ResizeImage.py]: Make this a global constant, please.

File: circle.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 machine:
   python:
-    version: 3.5.0
+    version: 2.7.11
```
Comment (TheKevJames) [inline circle.yml]: `2.7.12`?

**Review by TheKevJames — CHANGES_REQUESTED** [review]

File: circle.yml (hunk 2 above)
Comment (kyl191) [reply circle.yml]: Don't think Lambda is that up to date.

File: ResizeImage.py (hunk 1 above)
Comment (kyl191) [reply ResizeImage.py]: As in declare this when setting the log options?

Comment (TheKevJames) [reply ResizeImage.py]: Yeah, in global namespace. Just having vars like this all in one place is much easier to find that have to read every line.

File: circle.yml (hunk 2 above)
Comment (TheKevJames) [reply circle.yml]: 6 months is "up to date" but 12 months is not? Kay.

Comment (kyl191) [reply circle.yml]: Where's 6 months coming from?

In any case, I'm wrong: 
```
return sys.version
"2.7.12 (default, Sep  1 2016, 22:14:00) \n[GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]"```

Comment (TheKevJames) [reply circle.yml]: First result for googling "python 2.7.12 release date": https://www.python.org/downloads/release/python-2712/. Sorry, 5.5 months.

**Review by TheKevJames — APPROVED** [review]

**Pull request merged**